### ISSUE TITLE
Removal of deletion of the question categories produces unit test warning

### ISSUE THREAD

**Issue by dionysius** [frankkoch/moodle-mod_studentquiz#227]
The latest version generates a Risky test warning in phpunit and extra output:

```
vendor/bin/phpunit --colors "core_calendar_container_testcase" calendar/tests/container_test.php
Moodle 3.8.3 (Build: 20200511), 5afef2670e77c66c3671d7ef07e6514a77ac7835
Php: 7.3.17.1.18.04.1.1, pgsql: 10.12 (Ubuntu 10.12-0ubuntu0.18.04.1), OS: Linux 4.15.0-99-generic x86_64
PHPUnit 7.5.20 by Sebastian Bergmann and contributors.

................R<table class="generaltable">
<thead>
<tr>
<th class="header c0" style="" scope="col">Category</th>
<th class="header c1 lastcol" style="" scope="col">Action</th>
</tr>
</thead>
<tbody><tr class="">
<td class="cell c0" style="">top</td>
<td class="cell c1 lastcol" style="">This category has been deleted because, after deleting the course, its questions weren't used any more.</td>
</tr>
<tr class="lastrow">
<td class="cell c0" style="">Default for studentquiz 0</td>
<td class="cell c1 lastcol" style="">This category has been deleted because, after deleting the course, its questions weren't used any more.</td>
</tr>
</tbody>
</table>
.                                                18 / 18 (100%)

Time: 8.17 seconds, Memory: 91.00 MB

There was 1 risky test:

1) core_calendar_container_testcase::test_delete_module_delete_events
This test printed output: <table class="generaltable">
<thead>
<tr>
<th class="header c0" style="" scope="col">Category</th>
<th class="header c1 lastcol" style="" scope="col">Action</th>
</tr>
</thead>
<tbody><tr class="">
<td class="cell c0" style="">top</td>
<td class="cell c1 lastcol" style="">This category has been deleted because, after deleting the course, its questions weren't used any more.</td>
</tr>
<tr class="lastrow">
<td class="cell c0" style="">Default for studentquiz 0</td>
<td class="cell c1 lastcol" style="">This category has been deleted because, after deleting the course, its questions weren't used any more.</td>
</tr>
</tbody>
</table>

To re-run:
 vendor/bin/phpunit --colors "core_calendar_container_testcase" calendar/tests/container_test.php

OK, but incomplete, skipped, or risky tests!
Tests: 18, Assertions: 85, Risky: 1.
```

_Originally posted by @ewallah in https://github.com/frankkoch/moodle-mod_studentquiz/issues/173#issuecomment-626290135_

**Comment by dionysius** [comment]
Hi @timhunt, maybe you have a tip or comment for that, since you know moodle better.

Since removal of https://github.com/frankkoch/moodle-mod_studentquiz/commit/8160c62e4dcdea9c39fa8aa6e9c7f3777a67f8a4#diff-828043dc9f8d10ccd58ff8b96297794f you suggested in https://github.com/frankkoch/moodle-mod_studentquiz/issues/173#issuecomment-605505865 leads to the warning above. I'm afraid we have to add that part back in.

**Comment by timhunt** [comment]
So, what is going on here is that the kind of back-end functions that get tested in unit tests are generally not expected to generate output. Therefore, the unit test system warns if anything is output during the test run.

I realise that this is a core Moodle function doing the output, so we just have to live with that.

I was about to make a suggestion, however, it is strange. When I run the unit tests, I don't see this (no extra output.) I wonder why?

**Comment by timhunt** [comment]
Oh! I see. It is not a StudentQuiz test that is failing. It is one of the core_calendar unit tests. Hmm...

**Comment by dionysius** [comment]
Yeah correct, it is a separate test in moodle core provoking that. I can also see @ewallah's point. Some organisations might test over the full moodle before going to production with a version/plugin combination, which sounds very reasonable and clean to do so.

**Comment by timhunt** [comment]
Yes. I'm still thinking how we might fix it. It seems like a bug in Moodle core.

**Comment by timhunt** [comment]
So, I have concluded it is a bug in Moodle core, so I created MDL-68757. Except that, given the nature of the change, it is cleaning up old technical debt, rather than strictly fixing a bug, so I expect that it will only go into Moodle 3.9 or later.

Which means that, until then, we are probably just going to have to warn people that installing StudentQuiz causes that core test to be flagged as risky. Can we live with that?
